app/assets/javascripts/components/grouped_bar_chart.js.jsx

```diff
@@ -0,0 +1,220 @@
+    chartWidth = this.state.width,
+    bar, svg, scale, xAxis, labelWidth = 0;
+
+    // Zip the series data together (first values, second values, etc.)
+    var zippedData = [];
+    for (var i=0; i<data.labels.length; i++) {
+      for (var j=0; j<data.series.length; j++) {
+        zippedData.push(data.series[j].values[i]);
+      }
+    }
+
+    chartWidth -= (spaceForLabels + spaceForLegend);
+    chartHeight = barHeight * zippedData.length + gapBetweenGroups * data.labels.length + spaceForTicks;
+
+    if (data.labels.length==0) {
+	     chartHeight = 200;
+    }
+
+    max = d3.max(zippedData);
+
+    // Color scale
+    var color = d3.scale.category20();
+
+    var x = d3.scale.linear()
+    .domain([0, d3.max(zippedData)])
+    .range([0, chartWidth]);
+
+    var xAxisHeight=-chartHeight + 2*margin + axisMargin;
+    var	xAxis = d3.svg.axis()
+    .scale(x)
+    .tickSize(xAxisHeight)
+    .orient("bottom");
+
+    var y = d3.scale.linear()
+    .range([chartHeight + gapBetweenGroups, 0]);
+
+    var yAxis = d3.svg.axis()
+    .scale(y)
+    .tickFormat('')
+    .tickSize(0)
+    .orient("left");
+
+    // Specify the chart area and dimensions
+	  svg = d3.select(chart)
+	    .append("svg")
+	    .attr("width", "100%")
+	    .attr("height", chartHeight);
+		
+
+    // Create bars
+    var bar = svg.selectAll("g")
+    .data(zippedData)
+    .enter().append("g")
+    .attr("transform", function(d, i) {
+      return "translate(" + spaceForLabels + "," + (i * barHeight + gapBetweenGroups * (0.5 + Math.floor(i/data.series.length))) + ")";
+    });
+
+    // Create rectangles of the correct width
+    bar.append("rect")
+    .attr("fill", function(d,i) { return color(i % data.series.length); })
+    .attr("class", "bar")
+    .attr("width", x)
+    .attr("height", barHeight - 1);
+
+
+    // Add text label in bar
+    bar.append("text")
+    .attr("class", "horizontal-bar-value")
+    .attr("x", function(d) { return x(d) + 2; })
+    .attr("y", barHeight / 2)
+    .attr("fill", "black")
+    .attr("dy", ".35em")
+    .text(function(d) { return d; });
+
+    // Draw labels
+    var labels_offset=100;
+    bar.append("text")
+    .attr("class", "horizontal-bar-value")
+    .attr("x", function(d) { return - labels_offset; })
+    .attr("y", groupHeight / 2)
+    .attr("dy", ".35em")
+    .text(function(d,i) {
+      if (i % data.series.length === 0)
+      return data.labels[Math.floor(i/data.series.length)];
+      else
+      return ""}).each(function() {
+        labelWidth = spaceForLabels;
+      });
+
+
+    scale = d3.scale.linear()
+      .domain([0, max])
+      .range([0, chartWidth - margin*2 - labelWidth]);
+
+    xAxis11 = d3.svg.axis()
+      .scale(scale)
+      .tickSize(-chartHeight + 2*margin + axisMargin)
+      .orient("bottom");
+
+    svg.append("g")
+      .attr("class", "y axis")
+      .attr("transform", "translate(" + spaceForLabels + ", " + -gapBetweenGroups/2 + ")")
+      .call(yAxis);
+
+      // Draw legend
```

Comment:
Well, partially, look at the multi-line method calls in the lines 80-116 and 136-158 😉